PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskSDKL.h

```diff
@@ -0,0 +1,93 @@
+#ifndef AliAnalysisTaskSDKL_H
+#define AliAnalysisTaskSDKL_H
+
+// $Id$
+
+class TH1;
+class TH2;
+class TH3;
+class THnSparse;
+//class TTree;
+class TNtuple;
+class AliJetContainer;
+class AliParticleContainer;
+
+//#include "AliAnalysisTaskEmcal.h"
+#include "AliAnalysisTaskEmcalJet.h"
+#include "FJ_includes.h"
+
+namespace fastjet {
+    class PseudoJet;
+}
+
+struct split {
+    float z;
+    float r;
+    float m;
+    int sd_step;
+};
+
+class AliAnalysisTaskSDKL : public AliAnalysisTaskEmcalJet {
+ public:
+
+  AliAnalysisTaskSDKL();
+  AliAnalysisTaskSDKL(const char *name, Int_t const backgroption);
+  virtual ~AliAnalysisTaskSDKL();
+
+  void                        UserCreateOutputObjects();
+  void                        Terminate(Option_t *option);
+
+  static AliAnalysisTaskSDKL* AddTaskSoftDrop(
+    const char *ntracks            = "usedefault",
+    const char *njets              = "Jets",
+    const char *nrho               = "Rho",
+    Int_t       nCentBins          = 1,
+    Double_t    jetradius          = 0.4,
+    Double_t    jetptcut           = 1,
+    Double_t    jetareacut         = 0.6,
+    const char *type               = "EMCAL",
+    Int_t       backgroption       = 0,
+    Int_t       leadhadtype        = 0,
+    const char *taskname           = "AliAnalysisTaskSDKL"
+  );
+
+  static std::vector<split>   ReclusterFindHardSplits(AliEmcalJet *jet);
+  static std::vector<split>   ReclusterFindHardSplits(fastjet::PseudoJet const & jet);
+  static std::vector<split>   ReclusterFindHardSplits(std::vector <fastjet::PseudoJet> const & particles);
+  static std::vector<split>   FindHardSplits(fastjet::PseudoJet const & jet);
+
+  static std::vector<fastjet::PseudoJet> GetBackSubEvent(std::vector <fastjet::PseudoJet> const & full_event, Double_t & rho, Double_t & rho_sparse, Int_t opt = 0);
+
+  static Float_t              CalcMass(AliEmcalJet *jet);
+  static Float_t              CalcMass(fastjet::PseudoJet const & jet);
```

Comment:
Also the PseudoJet probably has already a function to provide the jet mass - no need for reimplementation

Reply:
Switched to PseudoJet::m(). However, it won't allow to easily switch the mass hypothesis on the fly.